homeassistant/components/sma/__init__.py

```diff
@@ -143,7 +144,15 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
     try:
         # Get updated device info
-        device_info = await sma.device_info()
+        sma_device_info = await sma.device_info()
+        device_info = DeviceInfo(
```

Comment:
Please only wrap the lines that can raise in the try...except block. We can move this line down.

Reply:
See #59821